### PR TITLE
52 validations and exception handling

### PR DESCRIPTION
**Pull request by BrianVega**
Business logic, validations, exception handling and 27 new more tests were added

### CODE REVIEW

File: src/main/java/org/ecommerce/validations/validators/OrderStatusValidator.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.ecommerce.validations.validators;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.ecommerce.enums.OrderStatus;
+import org.ecommerce.util.Enums;
+import org.ecommerce.validations.constraints.ValidOrderStatus;
+
+
+public class OrderStatusValidator implements ConstraintValidator<ValidOrderStatus, OrderStatus> {
```
Comment (algutierrez-griddynamics) [inline src/main/java/org/ecommerce/validations/validators/OrderStatusValidator.java:10]: Missing test

File: src/main/java/org/ecommerce/util/money/operations/UsdConverter.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * @param from String Currency code of the source
+     * @param to String Currency code of the expected output
+     * @param amount BigDecimal amount that will be converted
+     * @return The converted equivalence of the amount from source currency to expected currency
+     * @throws JsonProcessingException If {@link #apiCurrencyConverterService} fails while parsing
+     * the response to {@link ApiCurrencyConverterResponse} inside of {@link JsonParser}
+     */
+
+    public static BigDecimal convertAmountFromTo(String from, String to, BigDecimal amount) throws JsonProcessingException {
```
Comment (algutierrez-griddynamics) [inline src/main/java/org/ecommerce/util/money/operations/UsdConverter.java:28]: Missing the integration with order creation logic

File: src/main/java/org/ecommerce/validations/validators/OrderStatusValidator.java (hunk 1 above)
Comment (BrianVega) [reply src/main/java/org/ecommerce/validations/validators/OrderStatusValidator.java:10]: Here's the commit with the tests: https://github.com/algutierrez-griddynamics/e-commerce/commit/1c7c622da02d7ece8502d9665998de9a63928f75

File: src/main/java/org/ecommerce/util/money/operations/UsdConverter.java (hunk 2 above)
Comment (BrianVega) [reply src/main/java/org/ecommerce/util/money/operations/UsdConverter.java:28]: Here's the logic implemented https://github.com/algutierrez-griddynamics/e-commerce/commit/07f9404f0a60ccd720d38e7193067e3e6a868879

**Review by algutierrez-griddynamics — APPROVED** [review]

**Pull request merged**